### ISSUE TITLE
GH Actions conflict: an artifact with this name already exists on the workflow run

### ISSUE THREAD

**Issue by aanm** [cilium/cilium#31001]
```
Run actions/upload-artifact@5d5d22a31266ced268874388b861e4b58bb5c2f3
With the provided path, there will be 2 files uploaded
Artifact name is valid!
Root directory input is valid!
Error: Failed to CreateArtifact: Received non-retryable error: Failed request: (409) Conflict: an artifact with this name already exists on the workflow run
```

seen in https://github.com/cilium/cilium/actions/runs/8046107756/job/22000185630

**Comment by julianwiedmann** [comment]
I think this needs the backport from https://github.com/cilium/cilium/pull/30443

**Comment by aanm** [comment]
Closing as this is going to be fixed once #30443 is backported.
